src/hotspot/share/prims/nativeEntryPoint.cpp

```diff
@@ -26,10 +26,9 @@
 #include "runtime/interfaceSupport.inline.hpp"
 #include "code/vmreg.hpp"
 
-JVM_ENTRY(jlong, NEP_vmStorageToVMReg(JNIEnv* env, jclass _unused, jint type, jint index)) {
+JVM_ENTRY(jlong, NEP_vmStorageToVMReg(JNIEnv* env, jclass _unused, jint type, jint index))
```

Comment:
Still unclear why this is a JVM_ENTRY in the first place, if it simply switches back to native. To me there is some kind of code structuring issue here. If we don't need to be _thread_in_vm then we shouldn't be using a JVM_ENTRY.

Reply:
I missed this one in the round of fixing. Thanks for catching.

FWIW, I used JVM_ENTRY here since that was what our other existing code was already using, and it just worked, so I never gave it a second thought. I suspected there might be better ways of doing things, and we'd find out during code review. And here we are :)

But, I'm also having a bit of trouble understanding what the expectation here is. As you can see, this code is used as the backing implementation of a `native` method declared in `jdk.internal.invoke.NativeEntryPoint`. It translates the register representation we use on the Java side, into (one of) the VM's register representations (VMReg), which is then erased to a `long`. Later on C2 code extracts these values when it intrinsifies the native call. The code still works if I switch this to JVM_LEAF, which avoids doing the thread state transition. I assume it's preferred to use the JVM_* macros to guard code that uses VM internals (like this code does)? I could also try to roll my own macro/declaration if that makes more sense.